src/realm/object-store/results.cpp

```diff
@@ -310,8 +312,20 @@ util::Optional<Obj> Results::try_get(size_t row_ndx)
     validate_read();
     switch (m_mode) {
         case Mode::Empty:
-        case Mode::Collection:
             break;
+        case Mode::Collection: {
+            Mixed m = get_any(row_ndx);
+            if (m.is_type(type_Link) && m_table) {
+                return m_table->get_object(m.get<ObjKey>());
+            }
+            else if (m.is_type(type_TypedLink)) {
+                return m_realm->read_group().get_object(m.get_link());
+            }
+            else {
+                return Obj();
+            }
```

Comment:
> I am not sure that would be a good idea. Does this new behavior work for you? It will just return an unattached object on NULL.

Yes - that works fine for .NET - an `Optional<Obj>(none)` and `Obj()` behave quite identical anyway. My suggestion would be for this to only handle the `m.is_null()` case rather than the catch-all `else` here. It's probably a bug in the SDK if we call `results::try_get<Obj>` on a dictionary that contains `string` and the catch-all else will obscure that by returning a `null` value rather than `none`.

Reply:
Ok